Feature/unity/util/json manager

KikyoNanakusa opened the pull request:
# JsonManager.csを作成しました。
JsonManagerはLoadJsonとSaveJsonの2つの関数を持ちます。
LoadJsonはJSONファイルを読み込み、JsonData型のインスタンスを返します。
SaveJsonはJsonData型のインスタンスを引数として受け取り、JSONファイルとして保存します。

## 既知の問題
### JsonData Class
Jsonの内容を扱うためにJsonDataクラスを作成しましたが、命名に問題があると思っています。
なにかいい命名ないでしょうか。
### Jsonのパス
Jsonファイルが今のところどこに保存されるのか不明なのでダミーのJsonファイルを用意してそのパスにアクセスするようになています。これは、Swiftとマージする際に解決すると思います。

## JsonManagerの機能について
何か他に実装したほうが良い機能が思いつく方いたら提案してください

KikyoNanakusa committed:
feat[json]:Created JsonManager.cs and meta files.

KikyoNanakusa committed:
feat[json]:Added Read and Write json feature

KikyoNanakusa committed:
feat: Added exception handling on loading json

KikyoNanakusa committed:
feat: Added LoadJson and SaveJson functions in JsonManager.cs
Added JsonTestScene
Added JsonManagerTest.cs

KikyoNanakusa committed:
fix: Some typo fixed.
Some unnecessary comments deleted.

KinjiKawaguchi commented on Unity/Assets/Scripts/json/JsonManager.cs:
```diff
@@ -0,0 +1,119 @@
+
+//class名要検討 どんなデータなのか分かる名前だとよい
+[Serializable]
+public class JsonData
```
```suggestion
public class AppData
```
アプリに関するデータということで、ルートはAppDataでどうでしょうか

KinjiKawaguchi commented on Unity/Assets/Scripts/json/JsonManager.cs:
```diff
@@ -0,0 +1,119 @@
+    public int id;
+    public string name;
+    public string description;
+    public string createdAt;
```
```suggestion
    public string createdAt;　// YYYY-MM-DD/HH:MM:SS
```
といった具合でデータの不定な日時表現を表した方が保守性は上がりそうです。

KinjiKawaguchi commented on Unity/Assets/Scripts/json/JsonManager.cs:
```diff
@@ -0,0 +1,119 @@
+    public string name;
+    public string description;
+    public string createdAt;
+    public string updatedAt;
```
同上

KinjiKawaguchi commented on Unity/Assets/Scripts/json/JsonManager.cs:
```diff
@@ -0,0 +1,119 @@
+    public string description;
+    public string createdAt;
+    public string updatedAt;
+    public string dueDate;
```
同上こちらも時間持たせる感じでお願いします。

KinjiKawaguchi commented on Unity/Assets/Scripts/json/JsonManager.cs:
```diff
@@ -0,0 +1,119 @@
+public class JsonManager
+{
+    //まだダミーのjsonファイルのパスです。本番環境で変えます。
+    private const string JSON_PATH = "Assets/Resources/DummyJson.json";
```
このPathを決めてからMergeしましょうか

KinjiKawaguchi commented on Unity/Assets/Scripts/json/JsonManager.cs:
```diff
@@ -0,0 +1,119 @@
+            JsonData jsonData = JsonUtility.FromJson<JsonData>(jsonDataString);
+            return jsonData;
+        }
+        catch (Exception e)
```
エラーハンドリングはViewにも反映させる必要がありそうです。

KikyoNanakusa committed:
fix:Added dueDate to Task class
Change class name JsonData to AppData
Added CallErrorUI function
Deleted some comments
Added some comments

KikyoNanakusa commented:
TaskクラスにdueDateがなかったので追加しました。
クラス名をAppDataに変更しました。
エラーが発生した際にその旨を表示するUIを呼び出す関数を作成しました。
UI自体は未着手なので関数の中身は空です

KikyoNanakusa committed:
feat:Added JsonDataExchangeManager.cs

KinjiKawaguchi committed:
Merge branch 'develop' into feature/unity/util/jsonManager

KinjiKawaguchi commented:
ダミーファイルのパス云々はどうしますか？

KikyoNanakusa commented:
その辺いろいろ書き換えてからもう一回コミットします

KikyoNanakusa committed:
feat: Add path feature

KikyoNanakusa committed:
Merge branch 'feature/unity/util/jsonManager' of https://github.com/ShizuokaU-GreenTea-Programmers/TasQuest into feature/unity/util/jsonManager

KikyoNanakusa commented:
パス関連はまだ仕様が未定なので引数で指定できるようにしときました。

KinjiKawaguchi commented:
LGTM マージします。
